inspirehep/records/marshmallow/base.py

```diff
@@ -11,58 +11,62 @@
 from marshmallow.schema import Schema
 
 
-def schemaMetadataFactory(schema):
-    class InspireSchema(Schema):
-        id_ = fields.String(attribute="pid.object_uuid")
-        metadata = fields.Nested(schema, dump_only=True)
+class InspireEnvelopeSchema(RecordSchemaJSONV1):
+    """This schema is the envelope of the API.
 
-    return InspireSchema
+    Invenio API returns the following structure:
 
+    Examples:
+        {
+            id
+            metadata
+            created
+            updated
+        }
 
-class InspireBaseSchema(RecordSchemaJSONV1):
-    # FIXME: this should be renamed to ``uuid`` depends on ``inspire-next``
-    id_ = fields.String(attribute="pid.object_uuid")
+    Note:
+        In our case we need to enchance this with extra data.
+    """
 
+    id_ = fields.String(attribute="pid.object_uuid")
 
-class InspireBaseMetadataSchema(Schema):
-    _post_dumps = []
-
-    @post_dump(pass_original=True)
-    def process_post_dump_in_order(self, object_, original_data):
-        for dump_func in self._post_dumps:
-            object_ = dump_func(object_, original_data)
-        return strip_empty_values(object_)
 
+class InspireElasticSearchBaseSchema(Schema):
+    """ElasticSearch specific extra data."""
 
-class InspireESEnhancementSchema(Schema):
     _created = fields.DateTime(dump_only=True, attribute="created")
     _updated = fields.DateTime(dump_only=True, attribute="updated")
 
 
-class InspireIncludeAllFieldsSchemaMixin:
-    """Include all fields from a record."""
+class InspireBaseSchema(Schema):
```

Comment:
this sounds like this is a base schema for everything (incl. ES ones), but it doesn't seem so since `class InspireElasticSearchBaseSchema(Schema):`

Reply:
ElasticSearchBaseSchema contains fields that are required for ES only i.e. ``_created`` and ``_updated``

Reply:
Then let's do call this `RecordBaseSchema`, don't have a strong opinion about this though.

Reply:
renamed to just ``BaseSchema``